### PR TITLE
[Browser] Present errors to the user

### LINKED ISSUE #1354: Display all errors in the Realm Browser

### PR DESCRIPTION
**Pull request by segiddins**
Closes #1354.

\c @GreatApe @jpsim

### CODE REVIEW

File: tools/RealmBrowser/RealmBrowser/Classes/RLMRealmNode.m [hunk 1]
```diff
@@ -50,14 +50,15 @@ - (BOOL)connect:(NSError **)error
...
+
+    if (error && *error) {
         NSLog(@"Realm was opened with error: %@", *error);
     }
     else {
         _topLevelClasses = [self constructTopLevelClasses];    
     }
     
-    return error != nil;
+    return !(error && *error != nil);
```
Comment (alazier) [inline tools/RealmBrowser/RealmBrowser/Classes/RLMRealmNode.m]: This is incorrect in the case a user passes in nil. You need to pass in our own variable into realmWithPath and then set the user error after

Comment (segiddins) [inline tools/RealmBrowser/RealmBrowser/Classes/RLMRealmNode.m]: Fixed

**Comment by GreatApe** [comment]
Looks good to me. I take it these are all the realm related errors?

**Comment by segiddins** [comment]
Yes, these are the errors we pass into other methods.

**Comment by jpsim** [comment]
:+1:

**Pull request merged**